### PR TITLE
Source content updater UI

### PR DESCRIPTION
**Pull request by mannycolon**
#### Describe what your pull request addresses (Please include relevant issue numbers):
-

#### Please include detailed Test instructions for your pull request:
- Open tC and click the actions button then `Check for content updates`
- The accessing the internet alert should open
- Click ok and the checking for source content update dialog should pop up then either the failed alert or the actual list of resources that have updates should pop up
- select resources desired to update and click download
- then it should be downloaded or show an alert if it fails to do so

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/4929)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (27d5e18)](https://tc-builds.door43.org/translationCore-linux-x64-1.0.0-27d5e18.zip) is ready for download.
[macOS build 1.0.0 (27d5e18)](https://tc-builds.door43.org/translationCore-macos-x64-1.0.0-27d5e18.dmg) is ready for download.
[Windows x64 build 1.0.0 (27d5e18)](https://tc-builds.door43.org/translationCore-win-x64-1.0.0-27d5e18.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (27d5e18)](https://tc-builds.door43.org/translationCore-win-x32-1.0.0-27d5e18.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (55eef75)](https://tc-builds.door43.org/translationCore-linux-x64-1.0.0-55eef75.zip) is ready for download.
[macOS build 1.0.0 (55eef75)](https://tc-builds.door43.org/translationCore-macos-x64-1.0.0-55eef75.dmg) is ready for download.
[Windows x64 build 1.0.0 (55eef75)](https://tc-builds.door43.org/translationCore-win-x64-1.0.0-55eef75.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (55eef75)](https://tc-builds.door43.org/translationCore-win-x32-1.0.0-55eef75.setup.exe) is ready for download.

**Comment by PhotoNomad0** [comment]
Tests failing

File: src/js/actions/AlertModalActions.js [hunk 1]
```diff
@@ -21,17 +21,18 @@ export function openAlertDialog(alertMessage, loading) {
...
  * @param {String} alertMessage - message to be displayed inside the alert dialog.
- * @param {function} callback - message to be displayed inside the alert dialog.
+ * @param {function} callback - callback function that is trigger when clicking button1Text.
  * @param {String} button1Text - button text to show right button.
  * @param {String} button2Text - button text to show left of right button. (optional - if not present button is not added)
  * @param {String} buttonLinkText - button text to show on left link button. (optional - if not present button is not added)
  * @return {Object} action content.
  */
-export function openOptionDialog(alertMessage, callback, button1Text, button2Text, buttonLinkText = null) {
+export function openOptionDialog(alertMessage, callback, button1Text, button2Text, buttonLinkText = null, callback2) {
```
Comment (PhotoNomad0) [inline src/js/actions/AlertModalActions.js]: callback2 - probably add `= null`

**Review by PhotoNomad0 — COMMENTED** [review]
Source Looks good, just a minor change suggestion.  Will test now.

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (780a909)](https://tc-builds.door43.org/translationCore-linux-x64-1.0.0-780a909.zip) is ready for download.
[macOS build 1.0.0 (780a909)](https://tc-builds.door43.org/translationCore-macos-x64-1.0.0-780a909.dmg) is ready for download.
[Windows x64 build 1.0.0 (780a909)](https://tc-builds.door43.org/translationCore-win-x64-1.0.0-780a909.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (780a909)](https://tc-builds.door43.org/translationCore-win-x32-1.0.0-780a909.setup.exe) is ready for download.

File: src/js/helpers/sourceContentUpdatesHelpers.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+          localResourceList.push(localResource);
+        } else {
+          console.warn(`no such file or directory, ${pathToBibleManifestFile}`);
+        }
+      });
+
+      tHelpsResources.forEach(tHelpsId => {
+        const tHelpResource = path.join(tHelpsPath, tHelpsId);
+        const tHelpsLatestVersion = getLatestVersionInPath(tHelpResource);
```
Comment (PhotoNomad0) [inline src/js/helpers/sourceContentUpdatesHelpers.js:59]: Need to add check for null value being returned if no versions found.

File: src/js/helpers/sourceContentUpdatesHelpers.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    for (let i = 0; i < resourceLanguages.length; i++) {
+      const languageId = resourceLanguages[i];
+      const biblesPath = path.join(USER_RESOURCES_PATH, languageId, 'bibles');
+      const tHelpsPath = path.join(USER_RESOURCES_PATH, languageId, 'translationHelps');
+      const bibleIds = cleanReaddirSync(biblesPath);
+      const tHelpsResources = cleanReaddirSync(tHelpsPath);
+
+      bibleIds.forEach(bibleId => {
+        const bibleIdPath = path.join(biblesPath, bibleId);
+        const bibleLatestVersion = getLatestVersionInPath(bibleIdPath);
```
Comment (PhotoNomad0) [inline src/js/helpers/sourceContentUpdatesHelpers.js:37]: Need to add check for null value being returned if no versions found.

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (4eeea0c)](https://tc-builds.door43.org/translationCore-linux-x64-1.0.0-4eeea0c.zip) is ready for download.
[macOS build 1.0.0 (4eeea0c)](https://tc-builds.door43.org/translationCore-macos-x64-1.0.0-4eeea0c.dmg) is ready for download.
[Windows x64 build 1.0.0 (4eeea0c)](https://tc-builds.door43.org/translationCore-win-x64-1.0.0-4eeea0c.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (4eeea0c)](https://tc-builds.door43.org/translationCore-win-x32-1.0.0-4eeea0c.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (0cf0128)](https://tc-builds.door43.org/translationCore-linux-x64-1.0.0-0cf0128.zip) is ready for download.
[macOS build 1.0.0 (0cf0128)](https://tc-builds.door43.org/translationCore-macos-x64-1.0.0-0cf0128.dmg) is ready for download.
[Windows x64 build 1.0.0 (0cf0128)](https://tc-builds.door43.org/translationCore-win-x64-1.0.0-0cf0128.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (0cf0128)](https://tc-builds.door43.org/translationCore-win-x32-1.0.0-0cf0128.setup.exe) is ready for download.

**Pull request merged**